In [2]:
import pandas as pd 
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns   
from sklearn.preprocessing import LabelEncoder


In [3]:
df = pd.DataFrame(pd.read_excel("C:/Users/Acer/Desktop/fogo_cruzado2/exercicio1.xlsx"))

In [26]:
df_ocorrencia = df[df["ocorrencia_relacionada"].notnull()]
df_ocorrencia

id_ocorrencia              uf                   cidade  \
54            68913           Bahia         LAURO DE FREITAS   
74            68882            Pará                    BELEM   
123           68827      Pernambuco  CABO DE SANTO AGOSTINHO   
272           68665      Pernambuco                   OLINDA   
318           68616           Bahia                 SALVADOR   
...             ...             ...                      ...   
6524          61938            Pará                BENEVIDES   
6550          61910            Pará                    BELEM   
6629          61826            Pará               ANANINDEUA   
6669          61785  Rio de Janeiro           RIO DE JANEIRO   
6670          61784  Rio de Janeiro           RIO DE JANEIRO   

                                               endereco               bairro  \
54    Rua Lagoa dos Patos - Lagoa dos Patos - Ipitan...             IPITANGA   
74    R. São João - Tapanã, Belém - PA, 66825-110, B...               TAPANA   
123   Alto do Sol - R. da Saudade - Pte. dos Carvalh...  PONTE DOS CARVALHOS   
272   Av. Sen. Nilo de Souza Coelho - Ouro Preto, Ol...           OURO PRETO   
318   R. Raul Palmeiras, Guine - Pernambués, Salvado...           PERNAMBUES   
...                                                 ...                  ...   
6524  R. Dom Araújo, Itaquara - Distrito Murinin - B...                   NI   
6550  Passagem Ligação - Ocupação Gabriel Pimenta - ...          TERRA FIRME   
6629  Rua Belo Horizonte - Conj Olga Benário - Águas...         AGUAS LINDAS   
6669  Morro dos Macacos - Vila Isabel, Rio de Janeir...          VILA ISABEL   
6670  Morro dos Macacos - Vila Isabel, Rio de Janeir...          VILA ISABEL   

     sub_bairro                localidade   latitude  longitude  \
54          NaN           LAGOA DOS PATOS -12.894690 -38.304371   
74          NaN                       NaN  -1.353100 -48.460344   
123         NaN               ALTO DO SOL  -8.236902 -34.984315   
272         NaN                       NaN  -7.995144 -34.870100   
318         NaN                     GUINE -12.966920 -38.460439   
...         ...                       ...        ...        ...   
6524        NaN                   MURININ  -1.271364 -48.327602   
6550        NaN  OCUPAÇAO GABRIEL PIMENTA  -1.459897 -48.450825   
6629        NaN     CONJUNTO OLGA BENARIO  -1.399556 -48.392452   
6669        NaN         MORRO DOS MACACOS -22.911896 -43.255242   
6670        NaN         MORRO DOS MACACOS -22.911896 -43.255242   

         data_ocorrencia  ... civis_feridos_ocorrencia  \
54   2024-12-29 13:30:00  ...                      NaN   
74   2024-12-27 14:00:00  ...                      NaN   
123  2024-12-24 00:00:00  ...                      NaN   
272  2024-12-15 00:00:00  ...                      NaN   
318  2024-12-13 02:00:00  ...                      NaN   
...                  ...  ...                      ...   
6524 2024-01-11 23:40:00  ...                      NaN   
6550 2024-01-11 22:00:00  ...                      NaN   
6629 2024-01-07 10:00:00  ...                      NaN   
6669 2024-01-06 06:00:00  ...                      NaN   
6670 2024-01-06 04:30:00  ...                      NaN   

     agentes_mortos_ocorrencia agentes_feridos_ocorrencia chacina_ocorrencia  \
54                         NaN                        NaN                Sim   
74                         NaN                        1.0                Não   
123                        NaN                        NaN                Não   
272                        NaN                        NaN                Não   
318                        NaN                        NaN                Não   
...                        ...                        ...                ...   
6524                       NaN                        NaN                Não   
6550                       NaN                        NaN                Sim   
6629                       NaN                        

In [5]:
#visualizar todas as variáveis presentes no dataframe.
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6769 entries, 0 to 6768
Data columns (total 26 columns):
 #   Column                      Non-Null Count  Dtype         
---  ------                      --------------  -----         
 0   id_ocorrencia               6769 non-null   int64         
 1   uf                          6769 non-null   object        
 2   cidade                      6769 non-null   object        
 3   endereco                    6769 non-null   object        
 4   bairro                      6769 non-null   object        
 5   sub_bairro                  483 non-null    object        
 6   localidade                  2637 non-null   object        
 7   latitude                    6769 non-null   float64       
 8   longitude                   6769 non-null   float64       
 9   data_ocorrencia             6769 non-null   datetime64[ns]
 10  data de cadastro            6769 non-null   datetime64[ns]
 11  última atualização          6769 non-null   datetime64[n

Respondendo a pergunta 1.

In [ ]:
df["motivo_principal"].value_counts() 
 


motivo_principal
Homicidio/Tentativa          2806
Ação policial                1357
Não identificado             1082
Tentativa/Roubo               521
Disputa                       371
Operação policial             346
Briga                         110
Tiros a esmo                   41
Ataque a civis                 38
Tentativa/Roubo de cargas      19
Sequestro/Cárcere Privado      16
Arrastão                       16
Disparo Acidental              13
Suicídio                       11
Outro                          10
Tortura                        10
Rebelião/Fuga de presos         1
Tentativa/Roubo a banco         1
Name: count, dtype: int64

Para responder a segunda parte da pergunta 1, irei filtrar o dataframe para visualizar apenas as linhas onde 'motivo_principal' é 'Tiros a esmo'


In [ ]:
df_tiro_a_esmo = df[df['motivo_principal'] == 'Tiros a esmo']

# Exibindo o dataframe filtrado
df_tiro_a_esmo


id_ocorrencia              uf                   cidade  \
94            68857  Rio de Janeiro           RIO DE JANEIRO   
266           68671      Pernambuco                   OLINDA   
267           68670            Pará                    BELEM   
279           68658  Rio de Janeiro           RIO DE JANEIRO   
377           68551            Pará                    BELEM   
391           68536           Bahia                 SALVADOR   
425           68501      Pernambuco                   RECIFE   
581           68336      Pernambuco               CAMARAGIBE   
582           68335      Pernambuco               CAMARAGIBE   
615           68302      Pernambuco                   RECIFE   
866           68041  Rio de Janeiro           RIO DE JANEIRO   
974           67924            Pará                    BELEM   
998           67898  Rio de Janeiro           RIO DE JANEIRO   
1287          67586           Bahia                 SALVADOR   
1431          67431           Bahia                 SALVADOR   
1511          67342      Pernambuco                   GOIANA   
1586          67264  Rio de Janeiro          DUQUE DE CAXIAS   
1589          67261  Rio de Janeiro          DUQUE DE CAXIAS   
1616          67233  Rio de Janeiro           RIO DE JANEIRO   
1751          67091  Rio de Janeiro             BELFORD ROXO   
1906          66930           Bahia                 SALVADOR   
2022          66803           Bahia                 SALVADOR   
2401          66403           Bahia                 SALVADOR   
2753          66021           Bahia                 SALVADOR   
3074          65664  Rio de Janeiro                 MESQUITA   
3145          65584  Rio de Janeiro           RIO DE JANEIRO   
3539          65141  Rio de Janeiro           RIO DE JANEIRO   
3850          64799            Pará                    BELEM   
4110          64518  Rio de Janeiro           RIO DE JANEIRO   
4218          64399            Pará                    BELEM   
4482          64115  Rio de Janeiro           RIO DE JANEIRO   
4487          64109            Pará                    BELEM   
4608          63978  Rio de Janeiro           RIO DE JANEIRO   
4652          63930           Bahia                 SALVADOR   
4725          63853           Bahia                 SALVADOR   
5021          63543           Bahia                 SALVADOR   
5534          62992  Rio de Janeiro           RIO DE JANEIRO   
5856          62641           Bahia                 SALVADOR   
6452          62013  Rio de Janeiro           RIO DE JANEIRO   
6490          61972      Pernambuco  JABOATAO DOS GUARARAPES   
6711          61742           Bahia         LAURO DE FREITAS   

                                               endereco                bairro  \
94             Jacarezinho, Rio de Janeiro - RJ, Brasil           JACAREZINHO   
266           Varadouro, Olinda - PE, 53010-240, Brasil             VARADOURO   
267   Via Show - Rod. Augusto Montenegro - Parque Ve...          PARQUE VERDE   
279   Praça Camaçari - Ramos, Rio de Janeiro - RJ, B...                 RAMOS   
377   R. Magalhães Barata - Carananduba (Mosqueiro),...           CARANAMDUBA   
391                       Lobato, Salvador - BA, Brasil                LOBATO   
425   Comunidade do Detran - Iputinga, Recife - PE, ...              IPUTINGA   
581   Rua Governador Valadares - João Paulo II, Cama...         JOAO PAULO II   
582                      Viana, Camaragibe - PE, Brasil                 VIANA   
615   Comunidade do Coque - Ilha de Joana Bezerra, R...    ILHA JOANA BEZERRA   
866         Jardim Carioca, Rio de Janeiro - RJ, Brasil        JARDIM CARIOCA   
974   Rodovia Augusto Montenegro, 555 - Parque Verde...          PARQUE VERDE   
998    Vila do João - Maré, Rio de Janeiro - RJ, Brasil                  MARE   
1287    Buracão - Tancredo Neves, Salvador - BA, Brasil  BEIRU/TANCREDO NEVES   
1431  R. Domingos Requião - Liberdade, Salvador - BA...             LIBERDADE   
1511       Tejucupapo

In [8]:
df_acao_policial = df_tiro_a_esmo["acao_policial"]
df_acao_policial 

94      Não
266     Não
267     Não
279     Não
377     Não
391     Não
425     Não
581     Não
582     Não
615     Não
866     Não
974     Não
998     Não
1287    Não
1431    Não
1511    Não
1586    Não
1589    Não
1616    Não
1751    Sim
1906    Não
2022    Não
2401    Não
2753    Não
3074    Não
3145    Não
3539    Não
3850    Não
4110    Não
4218    Não
4482    Não
4487    Não
4608    Não
4652    Não
4725    Não
5021    Não
5534    Não
5856    Não
6452    Não
6490    Não
6711    Não
Name: acao_policial, dtype: object

Transformando a variável categórica em numérica para calcular a porcentagem.


In [ ]:
le = LabelEncoder()
df_acao_policial = le.fit_transform(df_acao_policial)

In [ ]:
df_acao_policial.mean()*100

np.float64(2.4390243902439024)

Respondendo a pergunta 4.

In [ ]:

mais_tiroteio = df_tiro_a_esmo.sort_values("uf")
mais_tiroteio["bairro"].value_counts()

bairro
MARE                    4
LIBERDADE               2
OLAVO BILAC             2
PARQUE VERDE            2
MUSSURUNGA              1
SUSSUARANA              1
PARIPE                  1
JARDIM SANTO INACIO     1
BEIRU/TANCREDO NEVES    1
VILA CANARIA            1
PERNAMBUES              1
LOBATO                  1
PEDREIRA                1
CAMPINA                 1
CARANAMDUBA             1
TERRA FIRME             1
VARADOURO               1
IPITANGA                1
CAPELINHA               1
IPUTINGA                1
JOAO PAULO II           1
TEJUCOPAPO              1
VIANA                   1
RAMOS                   1
INHAUMA                 1
ILHA JOANA BEZERRA      1
CAJUEIRO SECO           1
BRAS DE PINA            1
CAMPINHO                1
MADUREIRA               1
PACIENCIA               1
JARDIM CARIOCA          1
BANCO DE AREIA          1
HELIOPOLIS              1
JACAREZINHO             1
Name: count, dtype: int64

Pergunta 2: Quantas chacinas foram registradas em 2024? E qual estado registrou menos vítimas desse fenômeno? 


Foram registrados 85 casos de chacina no ano de 2024. Sendo o estado de Pernambuco o que registrou menos casos (5 casos. Aproximadamente 5,88%)

In [12]:

chacina =df[df["chacina_ocorrencia"] == "Sim"]
chacina

id_ocorrencia              uf                   cidade  \
54            68913           Bahia         LAURO DE FREITAS   
55            68912           Bahia         LAURO DE FREITAS   
185           68754           Bahia                 CAMACARI   
294           68643  Rio de Janeiro              NOVA IGUAÇU   
349           68584      Pernambuco  CABO DE SANTO AGOSTINHO   
...             ...             ...                      ...   
6550          61910            Pará                    BELEM   
6624          61831  Rio de Janeiro                  NITERÓI   
6629          61826            Pará               ANANINDEUA   
6669          61785  Rio de Janeiro           RIO DE JANEIRO   
6727          61724  Rio de Janeiro                  NITERÓI   

                                               endereco         bairro  \
54    Rua Lagoa dos Patos - Lagoa dos Patos - Ipitan...       IPITANGA   
55       Chafariz, Vila Praiana - Lauro de Freitas - BA   VILA PRAIANA   
185                  Santa Maria, Camaçari - BA, Brasil    SANTA MARIA   
294   R. Manoel da Silva Pereira - Prados Verdes, No...  PRADOS VERDES   
349      Pirapama, Cabo de Santo Agostinho - PE, Brasil       PIRAPAMA   
...                                                 ...            ...   
6550  Passagem Ligação - Ocupação Gabriel Pimenta - ...    TERRA FIRME   
6624  Morro do Cavalão - Icaraí, Niterói - RJ, 24230...         ICARAI   
6629  Rua Belo Horizonte - Conj Olga Benário - Águas...   AGUAS LINDAS   
6669  Morro dos Macacos - Vila Isabel, Rio de Janeir...    VILA ISABEL   
6727  Complexo do Viradouro - Viradouro, Niterói - R...      VIRADOURO   

     sub_bairro                localidade   latitude  longitude  \
54          NaN           LAGOA DOS PATOS -12.894690 -38.304371   
55          NaN                  CHAFARIZ -12.897171 -38.307954   
185         NaN                       NaN -12.718288 -38.340637   
294         NaN                       NaN -22.846356 -43.607509   
349         NaN                       NaN  -8.285247 -35.062217   
...         ...                       ...        ...        ...   
6550        NaN  OCUPAÇAO GABRIEL PIMENTA  -1.459897 -48.450825   
6624        NaN          MORRO DO CAVALAO -22.911117 -43.100751   
6629        NaN     CONJUNTO OLGA BENARIO  -1.399556 -48.392452   
6669        NaN         MORRO DOS MACACOS -22.911896 -43.255242   
6727        NaN     COMPLEXO DO VIRADOURO -22.905270 -43.086367   

         data_ocorrencia  ... civis_feridos_ocorrencia  \
54   2024-12-29 13:30:00  ...                      NaN   
55   2024-12-29 11:00:00  ...                      NaN   
185  2024-12-20 15:00:00  ...                      NaN   
294  2024-12-14 19:00:00  ...                      2.0   
349  2024-12-11 05:00:00  ...                      NaN   
...                  ...  ...                      ...   
6550 2024-01-11 22:00:00  ...                      NaN   
6624 2024-01-08 18:14:00  ...                      NaN   
6629 2024-01-07 10:00:00  ...                      NaN   
6669 2024-01-06 06:00:00  ...                      NaN   
6727 2024-01-03 11:44:00  ...                      1.0   

     agentes_mortos_ocorrencia agentes_feridos_ocorrencia chacina_ocorrencia  \
54                         NaN                        NaN                Sim   
55                         NaN                        NaN                Sim   
185                        NaN                        NaN                Sim   
294                        1.0                        NaN                Sim   
349                        NaN                        2.0                Sim   
...                        ...                        ...                ...   
6550                       NaN                        NaN                Sim   
6624                       NaN                        NaN                Sim   
6629                       NaN                        NaN                Sim   
6669                       NaN                

In [13]:
chacina["uf"].value_counts()

uf
Rio de Janeiro    47
Bahia             27
Pará               6
Pernambuco         5
Name: count, dtype: int64

Pergunta 3: Quantas pessoas foram vítimas de balas perdidas em 2024? E quantas vítimas teve o estado com mais casos? 

Ao total foram 13 vítimas de balas perdidas (8 vítimas fatais e 5 vítimas feridas.). O Estado do Rio de Janeiro teve o maior número de vítimas (5 vítimas), correspondendo a, aproximadamente, 38.46%.



In [14]:
df_bala_perdida= df[df['motivo_principal'] == 'Disparo Acidental']
df_bala_perdida

id_ocorrencia              uf                   cidade  \
170           68780           Bahia         LAURO DE FREITAS   
426           68500      Pernambuco                   OLINDA   
625           68292      Pernambuco                   RECIFE   
1370          67496  Rio de Janeiro                     MAGÉ   
1981          66845  Rio de Janeiro                NILÓPOLIS   
4539          64051           Bahia                 CAMACARI   
4811          63761            Pará                    BELEM   
5001          63563            Pará                    BELEM   
5189          63359  Rio de Janeiro              SÃO GONÇALO   
5770          62731  Rio de Janeiro           RIO DE JANEIRO   
6125          62360           Bahia            MADRE DE DEUS   
6395          62072  Rio de Janeiro                 MESQUITA   
6742          61709      Pernambuco  JABOATAO DOS GUARARAPES   

                                               endereco            bairro  \
170   Batalhão da Polícia de Choque, Recreio do Ipit...  RECREIO IPITANGA   
426   Estrada do Caenga (Estrada de Águas Compridas)...   AGUAS COMPRIDAS   
625   Tv. do Gusmão - São José, Recife - PE, 50020-5...          SAO JOSE   
1370                                   Magé, RJ, Brasil                NI   
1981  Avenida Carmela Dutra - Centro, Nilópolis - RJ...            CENTRO   
4539  R. da Linha - Parque Verde I, Camaçari - BA, 4...    PARQUE VERDE I   
4811  R. Gaspar Viana, 1259 - Campina, Belém - PA, 6...            REDUTO   
5001  Av. Arthur Bernardes, 5743 - Paracuri, Icoarac...          PARACURI   
5189                     Pita, São Gonçalo - RJ, Brasil            PITA I   
5770      Rua Noventa e Quatro, Galeão, Rio de janeiro             GALEAO   
6125                                 Madre de Deus - BA                NI   
6395  Condominio Beija Flor 5 - Avenida Eternidade -...         COSMORAMA   
6742  Alto sao Sebastião - Cavaleiro, Jaboatão dos G...         CAVALEIRO   

      sub_bairro                         localidade   latitude  longitude  \
170          NaN                                NaN -12.887661 -38.328760   
426          NaN                                NaN  -7.990334 -34.897757   
625          NaN                                NaN  -8.072340 -34.886559   
1370         NaN                                NaN -22.654681 -43.038174   
1981         NaN                                NaN -22.804354 -43.417946   
4539         NaN                                NaN -12.724848 -38.321692   
4811         NaN                                NaN  -1.444569 -48.492335   
5001         NaN  CONJUNTO HABITACIONAL PARACURI II  -1.316362 -48.486271   
5189         NaN                                NaN -22.847062 -43.074596   
5770  TUPIACANGA                                NaN -22.784364 -43.227303   
6125         NaN                                NaN -12.744399 -38.616369   
6395         NaN                                NaN -22.789959 -43.417801   
6742         NaN                 ALTO SAO SEBASTIAO  -8.092851 -34.967171   

         data_ocorrencia  ... civis_feridos_ocorrencia  \
170  2024-12-21 08:00:00  ...                      NaN   
426  2024-12-05 00:00:00  ...                      1.0   
625  2024-11-25 00:00:00  ...                      NaN   
1370 2024-10-13 10:00:00  ...                      1.0   
1981 2024-09-07 22:00:00  ...                      NaN   
4539 2024-04-18 22:00:00  ...                      NaN   
4811 2024-04-06 01:30:00  ...                      NaN   
5001 2024-03-28 11:00:00  ...                      NaN   
5189 2024-03-17 03:00:00  ...                      1.0   
5770 2024-02-22 13:36:00  ...                      NaN   
6125 2024-02-04 18:00:00  ...                      NaN   
6395 2024-01-19 15:00:00  ...                      NaN   
6742 2024-01-02 00:00:00  ...                      NaN   

     agentes_mortos_ocorrencia agentes_feridos_ocorrencia chacina_ocorrencia  \
170                        NaN                        1.0 

In [15]:
df_bala_perdida[["uf","agentes_feridos_ocorrencia", "civis_mortos_ocorrencia", "civis_feridos_ocorrencia","agentes_mortos_ocorrencia"]].value_counts("uf")

uf
Rio de Janeiro    5
Bahia             3
Pernambuco        3
Pará              2
Name: count, dtype: int64

In [16]:
df_vitimas = pd.read_excel("C:/Users/Acer/Desktop/fogo_cruzado2/exercicio1_vitimas.xlsx")

In [17]:
df_vitimas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5936 entries, 0 to 5935
Data columns (total 39 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   id_ocorrencia           5936 non-null   int64         
 1   uf                      5936 non-null   object        
 2   endereco                5936 non-null   object        
 3   cidade                  5936 non-null   object        
 4   bairro                  5936 non-null   object        
 5   sub_bairro              450 non-null    object        
 6   localidade              2003 non-null   object        
 7   latitude                5936 non-null   float64       
 8   longitude               5936 non-null   float64       
 9   data_ocorrencia         5936 non-null   datetime64[ns]
 10  hora_ocorrencia         5936 non-null   object        
 11  presenca_agentes        5936 non-null   object        
 12  acao_policial           5936 non-null   object  

In [18]:
df_idade = df_vitimas[df_vitimas["faixa_etaria"] == "Criança"]
df_idade


id_ocorrencia              uf  \
195           68742           Bahia   
196           68734  Rio de Janeiro   
202           68729           Bahia   
219           68708           Bahia   
301           68611  Rio de Janeiro   
320           68588           Bahia   
321           68588           Bahia   
322           68588           Bahia   
331           68583  Rio de Janeiro   
391           68509  Rio de Janeiro   
393           68509  Rio de Janeiro   
635           68270  Rio de Janeiro   
672           68227  Rio de Janeiro   
886           67976  Rio de Janeiro   
889           67971            Pará   
985           67861  Rio de Janeiro   
1213          67592  Rio de Janeiro   
1367          67410           Bahia   
1678          67071  Rio de Janeiro   
1813          66932  Rio de Janeiro   
1889          66843  Rio de Janeiro   
1940          66779           Bahia   
2406          66243      Pernambuco   
2765          65733  Rio de Janeiro   
2979          65407  Rio de Janeiro   
3126          65227  Rio de Janeiro   
3389          64889  Rio de Janeiro   
3420          64840      Pernambuco   
3437          64800  Rio de Janeiro   
4221          63800  Rio de Janeiro   
4225          63800  Rio de Janeiro   
4498          63469            Pará   
4574          63363           Bahia   
4595          63344  Rio de Janeiro   
4687          63244  Rio de Janeiro   
5079          62751      Pernambuco   
5080          62751      Pernambuco   
5182          62621      Pernambuco   
5250          62526  Rio de Janeiro   
5377          62372           Bahia   
5390          62355  Rio de Janeiro   
5421          62320  Rio de Janeiro   
5697          61972      Pernambuco   
5869          61751  Rio de Janeiro   

                                               endereco  \
195        BR-324 - Águas Claras, Salvador - BA, Brasil   
196   Morro São João- Engenho Novo, Rio de Janeiro -...   
202   R. Francisco Ferraro - Nazaré, Salvador - BA, ...   
219                  R. Lindóia, Itinga - Salvador - BA   
301   Morro da Fé - Complexo da Penha, Rio de Janeir...   
320   R. Estácio de Sá - Campinas de Pirajá, Salvado...   
321   R. Estácio de Sá - Campinas de Pirajá, Salvado...   
322   R. Estácio de Sá - Campinas de Pirajá, Salvado...   
331   Rua Guaraúna - Vicente de Carvalho, Rio de Jan...   
391          Gardênia Azul, Rio de Janeiro - RJ, Brasil   
393          Gardênia Azul, Rio de Janeiro - RJ, Brasil   
635   Favela da Carobinha - Campo Grande, Rio de Jan...   
672   Santa Cruz da Serra, Duque de Caxias - RJ, Brasil   
886   Conjunto da Marinha - Bangu, Rio de Janeiro - ...   
889   Tv. Vileta - Pedreira, Belém - PA, 66087-423, ...   
985   Estrada Miguel Pereira - Nova Belém, Japeri - ...   
1213  Bela Vista - Nova Aurora, Belford Roxo - RJ, B...   
1367  Teotonio Vilela - Fazenda Coutos, Salvador - B...   
1678       Vila Santa Teresa, Belford Roxo - RJ, Brasil   
1813     Rua da Canastra - Alecrim, Japeri - RJ, Brasil   
1889  Comunidade do JJ- Tomás Coelho, Rio de Janeiro...   
1940      Av. Fernandes da Cunha, Mares - Salvador - BA   
2406  UR-04 - R. Sd. Virgílio Lúcio - COHAB, Recife ...   
2765                Pilar, Duque de Caxias - RJ, Brasil   
2979  Morro do Chaves - Barros Filho, Rio de Janeiro...   
3126  Morro da Mineira - Catumbi, Rio de Janeiro - R...   
3389  Favela Cavalo de Aço - Senador Camará, Rio de ...   
3420  Hospital Barão de Lucena - Avenida Caxangá - I...   
3437  Complexo Mangueirinha, Centenario, Duque de Ca...   
4221       Km 49 Boa Esperança, Seropédica - RJ, Brasil   
4225       Km 49 Boa Esperança, Seropédica - RJ, Brasil   
4498  R. Cap. Tomé Serrão - Barcarena, PA, 68445-000...   
4574                 Saramandaia, Salvador - BA, Brasil   
4595  Estrada do Baldeador - Baldeador, Niterói - RJ...   
4687  R. Cambuci, Vila São João,  Vilar dos Teles, S...   
5079  R. da Biquinha - Pilar, Ilha de Itamaracá - PE...   
5080  R. da Biquinha - Pilar, Ilha de Itamaracá - PE...   
5182  R.

Respondendo a Pergunta 5.


In [19]:
df_local = df_idade[df_idade["local"] == "Residência" ] 
df_local_filtered = df_local[df_local["circunstancias_vitima"] == "Bala perdida"]
df_local_filtered

id_ocorrencia              uf  \
219           68708           Bahia   
3126          65227  Rio de Janeiro   
5377          62372           Bahia   
5869          61751  Rio de Janeiro   

                                               endereco              cidade  \
219                  R. Lindóia, Itinga - Salvador - BA            SALVADOR   
3126  Morro da Mineira - Catumbi, Rio de Janeiro - R...      RIO DE JANEIRO   
5377                          Mussurunga, Salvador - BA            SALVADOR   
5869   Coelho da Rocha, São João de Meriti - RJ, Brasil  SÃO JOÃO DE MERITI   

               bairro sub_bairro                                 localidade  \
219            ITINGA        NaN                                        NaN   
3126          CATUMBI        NaN  MORRO DA MINEIRA (COMPLEXO DO SAO CARLOS)   
5377       MUSSURUNGA        NaN                                        NaN   
5869  COELHO DA ROCHA        NaN                                        NaN   

       latitude  longitude data_ocorrencia  ... status_servico  \
219  -12.901256 -38.347508      2024-12-18  ...  Não se aplica   
3126 -22.918444 -43.198411      2024-06-17  ...  Não se aplica   
5377 -12.914235 -38.376199      2024-02-04  ...  Não se aplica   
5869 -22.776883 -43.384963      2024-01-01  ...  Não se aplica   

     observacoes_gerais status_politico posicao_politico partido_político  \
219                 NaN   Não se aplica    Não se aplica              NaN   
3126                NaN   Não se aplica    Não se aplica              NaN   
5377                NaN   Não se aplica    Não se aplica              NaN   
5869                NaN   Não se aplica    Não se aplica              NaN   

     unidade_agente posicao_agente coorporacao_agente  status_agente  \
219             NaN  Não se aplica      Não se aplica  Não se aplica   
3126            NaN  Não se aplica      Não se aplica  Não se aplica   
5377            NaN  Não se aplica      Não se aplica  Não se aplica   
5869            NaN  Não se aplica      Não se aplica  Não se aplica   

     tipo_pessoa  
219        Civíl  
3126       Civíl  
5377       Civíl  
5869       Civíl  

[4 rows x 39 columns]

In [20]:
df_local_filtered["situacao"]

219      Morta
3126    Ferida
5377    Ferida
5869    Ferida
Name: situacao, dtype: object

In [21]:
df_local["circunstancias_vitima"]

219          Bala perdida
3126         Bala perdida
5079    Sem identificação
5080    Sem identificação
5182    Sem identificação
5377         Bala perdida
5869         Bala perdida
Name: circunstancias_vitima, dtype: object

In [22]:
df_qualificacao = df_vitimas[df_vitimas["qualificacoes_vitima"] == "funcionários de limpeza urbana" ]
df_qualificacao 

Empty DataFrame
Columns: [id_ocorrencia, uf, endereco, cidade, bairro, sub_bairro, localidade, latitude, longitude, data_ocorrencia, hora_ocorrencia, presenca_agentes, acao_policial, unidades_policiais, motivo_principal, motivos_complementares, outros_recortes, ocorrencia_relacionada, id_vitima, tipo, situacao, data_morte, idade, faixa_etaria, raca, genero, local, circunstancias_vitima, qualificacoes_vitima, status_servico, observacoes_gerais, status_politico, posicao_politico, partido_político, unidade_agente, posicao_agente, coorporacao_agente, status_agente, tipo_pessoa]
Index: []

[0 rows x 39 columns]

In [23]:
df_vitimas["qualificacoes_vitima"].value_counts()

qualificacoes_vitima
Não identificado                             4706
PPTE                                          498
Outros                                        315
Agente de segurança                           203
Mototaxista                                    46
Motorista de aplicativo                        43
Entregador / Motoboy                           24
Outros,PPTE                                    21
Gestante                                       10
Político                                        8
Não identificado,Outros                         7
Liderança religiosa                             5
Entregador / Motoboy,Mototaxista                4
Vendedor ambulante                              4
Rifeiro                                         3
Motorista de aplicativo,Outros                  3
Gestante,Outros                                 3
Não identificado,Político                       3
Mototaxista ,Outros                             3
Agente de segurança,Outros   

Respondendo a pergunta 6

In [24]:
df[df["id_ocorrencia"] == 67820]

id_ocorrencia          uf  cidade  \
1069          67820  Pernambuco  OLINDA   

                                               endereco            bairro  \
1069  Via do Contôrno - Jardim Atlântico, Olinda - P...  JARDIM ATLANTICO   

     sub_bairro localidade  latitude  longitude data_ocorrencia  ...  \
1069        NaN        NaN -7.972683 -34.846196      2024-10-30  ...   

     civis_feridos_ocorrencia agentes_mortos_ocorrencia  \
1069                      NaN                       NaN   

     agentes_feridos_ocorrencia chacina_ocorrencia unidades_policiais  \
1069                        NaN                Não                NaN   

         motivo_principal  motivos_complementares  outros_recortes  \
1069  Homicidio/Tentativa                     NaN              NaN   

      transporte_interrompido ocorrencia_relacionada  
1069                      Não                    NaN  

[1 rows x 26 columns]